www/js/app.js

```diff
@@ -760,6 +760,9 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
         if (contentInjectionMode === 'serviceworker') {
             // In ServiceWorker mode, we simply set the iframe src.
             // (reading the backend is handled by the ServiceWorker itself)
+            var encodedUrl = dirEntry.url.replace(/([^/]+)/g, function(_p0, part) {
```

Comment:
Maybe this piece of code could be moved closer to where this variable is used?
A comment should be added to explain what it does.
And the _p0 parameter should be renamed to something more meaningful (like `matchedSubstring`). In this particular case, the first capture group is the entire matched substring, so you could probably remove the capture group, and use this first parameter instead, like : 
```suggestion
            var encodedUrl = dirEntry.url.replace(/[^/]+/g, function(matchedSubstring) {
                return encodeURIComponent(matchedSubstring);
            });
```

Reply:
OK, yes this looks good. I hadn't noticed that I was selecting a part that is actually the same as the fully matched string.

Reply:
Regarding moving the code closer to where the variable is used, I put it where it is so that it wouldn't have to be run on article load. Any closer would involve putting it inside the onload function, and it seems to me there's already quite a lot going on at that time. Maybe it makes no difference, but instinctively that was my reason.

Reply:
OK you can keep it here, but please add a comment above to explain why we do that